discovery-provider/src/tasks/index.py

```diff
@@ -365,307 +499,106 @@ def index_blocks(self, db, blocks_list):
         update_ursm_address(self)
         block = blocks_list[i]
         block_index = num_blocks - i
-        block_number = block.number
-        block_hash = block.hash
-        block_timestamp = block.timestamp
-        latest_block_timestamp = block_timestamp
+        block_number, block_hash = itemgetter("number", "hash")(block)
         logger.info(
             f"index.py | index_blocks | {self.request.id} | block {block.number} - {block_index}/{num_blocks}"
         )
         challenge_bus: ChallengeEventBus = update_task.challenge_event_bus
-        with db.scoped_session() as session, challenge_bus.use_scoped_dispatch_queue():
-            current_block_query = session.query(Block).filter_by(is_current=True)
-
-            # Without this check we may end up duplicating an insert operation
-            block_model = Block(
-                blockhash=web3.toHex(block.hash),
-                parenthash=web3.toHex(block.parentHash),
-                number=block.number,
-                is_current=True,
-            )
-
-            # Update blocks table after
-            assert (
-                current_block_query.count() == 1
-            ), "Expected single row marked as current"
-
-            previous_block = current_block_query.first()
-            previous_block.is_current = False
-            session.add(block_model)
-
-            user_factory_txs = []
-            track_factory_txs = []
-            social_feature_factory_txs = []
-            playlist_factory_txs = []
-            user_library_factory_txs = []
-            user_replica_set_manager_txs = []
-
-            tx_receipt_dict = fetch_tx_receipts(self, block.transactions)
-
-            # Sort transactions by hash
-            sorted_txs = sorted(block.transactions, key=lambda entry: entry["hash"])
 
+        with db.scoped_session() as session, challenge_bus.use_scoped_dispatch_queue():
             skip_tx_hash = save_and_get_skip_tx_hash(session, redis)
```

Comment:
I'd love to rename this method something a bit more useful--
maybe

```
tx_hash_to_skip = get_indexing_error_with_consensus(redis)
if tx_hash_to_skip:
  save_skipped_tx_hash(session, tx_hash_to_skip)  
```

Reply:
This is a great name and helps my own understanding. I'll rename it! It totally makes sense to break out the contents of that fn that way so it's easier to understand.

Reply:
Minor nit: I'm torn on this...`get_*indexing_error*_with_consensus` - but it returns a transaction hash, not an error. Assigning the return value to a variable named `tx_hash_to_skip` alleviates this confusion here, but worth considering? Maybe `get_indexing_error_tx_with_consensus`

Reply:
i changed this to `get_tx_hash_to_skip`!